js/web/lib/wasm/jsep/webgpu/ops/binary-op.ts

```diff
@@ -167,8 +176,10 @@ const createBinaryOpProgramInfo =
 const createBinaryOpProgramInfoLoader =
     (inputs: readonly TensorView[], name: string, funcCall: BinaryFunctionCall, additionalImplementation?: string,
      cacheKey?: string): ProgramInfoLoader => {
+      validateInputs(inputs);
+      const cacheKeyWithType = cacheKey === undefined ? `${inputs[0].dataType}` : cacheKey + `_${inputs[0].dataType}`;
```

Comment:
if we merge this PR: https://github.com/microsoft/onnxruntime/pull/17176, this line can revert.